src/elf_program_header_table.lem

```diff
@@ -160,31 +161,39 @@ let allowable_permissions_of_permission m =
     return 7
   else
     fail "exact_permission_of_permission: invalid permission flag"
-    
-(** [string_of_elf_segment_permissions m] produces a string-based representation
-  * of an ELF segment's permission field.
-  * TODO: expand this as is needed by the validation tests.
+
+(** [elf64_interpreted_program_header_flags w] extracts a boolean triple of flags
+  * from the flags field of an interpreted segment.
   *)
-val string_of_elf_segment_permissions : natural -> string
-let string_of_elf_segment_permissions m =
+val parse_elf_segment_permissions : natural -> (bool * bool * bool)
+let parse_elf_segment_permissions m =
```

Comment:
I've done this because it doesn't seem like there are bit manipulation operations on naturals.